File: Boca/Q7/src/Ponto.java
public class Ponto {
    
    private double x;
    private double y;

    public void setX(double x) {
        this.x = x;
    }

    public void setY(double y) {
        this.y = y;
    }
    

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public static double distance(Ponto P_1, Ponto P_2) {
        double diff_X = Math.abs(P_2.getX() - P_1.getX());
        double diff_Y = Math.abs(P_2.getY() - P_1.getY());
        return Math.sqrt(Math.pow(diff_X, 2) + Math.pow(diff_Y, 2));
    }

}

File: Boca/CPP/01/src/Ponto.cpp
#include "Ponto.hpp"
#include <cmath>

Ponto::Ponto(double _x, double _y) {
    x = _x;
    y = _y;
}

double Ponto::distancia(const Ponto& ponto) const {
    double dx = x - ponto.x;
    double dy = y - ponto.y;
    return std::sqrt(dx * dx + dy * dy);
}
